src/toil/bus.py

```diff
@@ -741,12 +741,12 @@ def replay_message_bus(path: str) -> Dict[str, JobStatus]:
 
     return job_statuses
 
-def gen_message_bus_path() -> str:
+def gen_message_bus_path(tmpdir_prefix: Optional[str] = None) -> str:
     """
     Return a file path in tmp to store the message bus at.
     Calling function is responsible for cleaning the generated file.
     """
-    fd, path = tempfile.mkstemp()
+    fd, path = tempfile.mkstemp(prefix=tmpdir_prefix)
```

Comment:
It looks like using an absolute path as a prefix actually does work. We might need a comment asserting this; I thought this was wrong when I looked at it, and the docs don't seem to promise that this works.

What if the `--tmpdir-prefix` option is a relative path? It looks like `mkstemp` interprets a non-absolute prefix as relative to its chosen temp directory, not relative to the current directory. Will something turn it into an absolute path before we get here? If so, we should cover that in the docstring.

Reply:
If the prefix passed into `mkstemp` is a relative path, it looks like it will try to prefix that into the filename itself, choosing a valid tmp directory option
```python
>>> tmpdir_prefix = "test"
>>> fd, path = tempfile.mkstemp(prefix=tmpdir_prefix)
>>> path
'/tmp/testmo78tq3y'
```

Reply:
cwltool handles relative paths by creating the tmp file relative to the current working directory. I can probably mimic that.